### ISSUE TITLE
you can inject hulk dna in yourself after activating a desword to have hulk with a desword

### ISSUE THREAD

**Issue by Fikou** [tgstation/tgstation#70476]
## Reproduction:

get a desword
activate the desword in your hand
have someone inject hulk in you/use tk on the genetics console to give yourself hulk
free stun immunity + projectile immunity

**Comment by Guestratem** [comment]
Are there any other weapons/items that require the same treatment? or is it just the desword?

**Comment by msgerbs** [comment]
desword seems to have a snowflake check for hulkness so probably not. This also most definitely not how we should be checking for hulkness either...

https://github.com/tgstation/tgstation/blob/043ee07a636e0c6595fe1414ff119d3cdfd87b88/code/game/objects/items/dualsaber.dm#L51
`/obj/item/dualsaber/proc/on_wield(obj/item/source, mob/living/carbon/user)
	if(user?.has_dna())
		if(user.dna.check_mutation(/datum/mutation/human/hulk))
			to_chat(user, span_warning("You lack the grace to wield this!"))
			return COMPONENT_TWOHANDED_BLOCK_WIELD
	w_class = w_class_on
	hitsound = 'sound/weapons/blade1.ogg'
	START_PROCESSING(SSobj, src)
	set_light_on(TRUE)`

This whole check should probably be moved to `/datum/component/two_handed` as an optional flag, with a signal for becoming hulk (or more generically, acquiring a mutation?)

**Comment by Guestratem** [comment]
Yeah, it's only on wield check I assume you mean something similar to this?

![image](https://user-images.githubusercontent.com/57818568/196030852-05c690e9-a35c-4cb4-89c8-968495b1b075.png)

**Comment by msgerbs** [comment]
Yeah that looks vaguely like what I was thinking of. You would want to check for TRAIT_HULK rather than checking the DNA directly though.

**Comment by Guestratem** [comment]
I would have this done by now, but my drive corrupted so badly that I had to replace it

**Comment by Guestratem** [comment]
In testing i realise that if you try to hit someone with the desword it makes you drop the sword

**Comment by msgerbs** [comment]
So not an exploit, just a noob trap.

**Comment by Guestratem** [comment]
yeah, still bad, stun immune and immune to being shot

**Comment by Shadowflame909** [comment]
a cruel (Funny) way to deal with this is to have hulks crush all deswords into dust inhand. No matter how you get your hulk de-sword. It will never be encouraged!
